browser/importer/extensions_import_helpers.cc

```diff
@@ -5,92 +5,330 @@
+      extensions::value_store_util::ModelType::EXTENSION, extension_id,
+      target_store_factory);
+
+  auto target_store = extensions::value_store_util::CreateSettingsStore(
+      extensions::settings_namespace::LOCAL,
+      extensions::value_store_util::ModelType::EXTENSION, extension_id,
+      target_store_factory);
+
+  const auto result = target_store->Set(value_store::ValueStore::DEFAULTS,
+                                        settings.PassSettings());
+  return result.status().ok();
+}
+
+bool ImportIndexedDBExtensionSettings(const std::string& extension_id,
+                                      const base::FilePath& source_profile,
+                                      const base::FilePath& target_profile) {
+  constexpr char kIndexedDBDir[] = "IndexedDB";
+
+  const base::FilePath pattern = base::FilePath::FromASCII(
+      base::StrCat({"chrome-extension_", extension_id, "_*indexeddb*"}));
+
+  base::FileEnumerator target_enumerator(
+      target_profile.AppendASCII(kIndexedDBDir), false,
+      base::FileEnumerator::DIRECTORIES, pattern.value());
+  // Clear the target profile.
+  target_enumerator.ForEach(
+      [](const base::FilePath& path) { base::DeletePathRecursively(path); });
+
+  base::FileEnumerator source_enumerator(
+      source_profile.AppendASCII(kIndexedDBDir), false,
+      base::FileEnumerator::DIRECTORIES, pattern.value());
+  // Copy settings.
+  source_enumerator.ForEach([&target_profile,
+                             &kIndexedDBDir](const base::FilePath& path) {
+    base::CopyDirectory(
+        path, target_profile.AppendASCII(kIndexedDBDir).Append(path.BaseName()),
+        true);
+  });
+
+  return true;
+}
+
+bool ImportExtensionSettings(const std::string& extension_id,
+                             const base::FilePath& source_profile,
+                             const base::FilePath& target_profile) {
+  if (!ImportLocalExtensionSettings(extension_id, source_profile,
+                                    target_profile)) {
+    return false;
+  }
+  if (!ImportIndexedDBExtensionSettings(extension_id, source_profile,
+                                        target_profile)) {
+    return false;
+  }
+  return true;
+}
+
+ImportingExtension::ImportingExtension() = default;
+ImportingExtension::ImportingExtension(ImportingExtension&&) = default;
+ImportingExtension& ImportingExtension::operator=(ImportingExtension&&) =
+    default;
+
+ImportingExtension::~ImportingExtension() {
+  if (installer) {
+    installer->AbortInstall();
+  }
+}
+
+ExtensionsImporter::ExtensionsImporter(const base::FilePath& source_profile,
+                                       Profile* target_profile)
+    : source_profile_(source_profile), target_profile_(target_profile) {
+  CHECK(target_profile_);
+  CHECK_NE(source_profile_, target_profile_->GetPath());
+}
+
+ExtensionsImporter::~ExtensionsImporter() = default;
+
+// static
+base::RepeatingCallback<ExtensionImportStatus(const std::string& extension_id)>&
```

Comment:
What about making this non-static to clean this automatically?
Reseting it after using in tests looks so-so.

Reply:
There is only one test that uses it. I don't want to do more things more complicated here.